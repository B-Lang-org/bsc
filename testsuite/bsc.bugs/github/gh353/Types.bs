package Types where

-- Classic now treats these two types differently

data MyList1 a = Nil | Cons a (MyList1 a)

data MyList2 a = Nil | Cons { head :: a; tail :: (MyList2 a) }

-- When the tag's arguments are given names, then it is visible to the
-- user that a struct was created (but the name of the struct is not
-- known).
