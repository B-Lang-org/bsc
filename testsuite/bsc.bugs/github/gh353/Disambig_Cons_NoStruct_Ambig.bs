package Disambig_Cons_NoStruct_Ambig where

import Types

-- -----

fn1 :: MyList2 Bool
fn1 = let res = Cons { head = True; tail = Nil; };
      in res
