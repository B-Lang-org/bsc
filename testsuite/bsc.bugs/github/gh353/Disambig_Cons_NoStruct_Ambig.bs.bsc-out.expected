checking package dependencies
compiling Disambig_Cons_NoStruct_Ambig.bs
Error: "Disambig_Cons_NoStruct_Ambig.bs", line 8, column 16: (T0019)
  Constructor `Cons' is not disambiguated by type `_tctyvar1003'
